bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBaseThingHandler.java

```diff
@@ -107,14 +107,15 @@ protected void updateChannel(String channelId) {
             logger.debug("Value retrieved for channel '{}' was null. Can't update.", channelId);
             return;
         }
+        logger.trace("updateChannel channelId: {} value = {}", channelId, value);
 
         State state = null;
         if (value instanceof BigDecimal) {
             state = new DecimalType((BigDecimal) value);
         } else if (value instanceof Integer) {
             state = new DecimalType(BigDecimal.valueOf(((Integer) value).longValue()));
         } else if (value instanceof Double) {
-            state = new DecimalType((double) value);
+            state = new DecimalType((Double) value);
```

Comment:
You should be able to check `instanceof Number` and cast to `(Number)` for all of these cases.

Reply:
`QuantityType` is also a sub-class of `Number`.

Reply:
Thanks again. I didn't realise that. I've pushed another one that moved QuantityType before Number.